api/Controllers/OfferController.cs

```diff
@@ -77,5 +80,113 @@ public async Task<IActionResult> DeleteOffer(Guid offerId)
 
             return Ok(await _offerService.DeleteOffer(offerId));
         }
+
+        //images
+        [HttpGet("image/{offerslug}"), Authorize(Roles = "User")]
+        public async Task<IActionResult> GetOfferImages(string offerslug)
+        {
+            var offer = await _offerService.GetOfferBySlug(offerslug, Guid.Empty);
+
+            if (offer == null)
+            {
+                return NotFound("offer didn't exist");
+            }
+
+            if (!offer.OfferImages.Any())
+            {
+                return NotFound("offer didn't have uploaded images yet");
+            }
+
+            var offerImages = await _fileService.GetOfferImagesByOfferSlug(offerslug);
+
+            foreach (var offerImage in offerImages)
+            {
+                if (offerImage.ImageBytes.Length < 1)
+                {
+                    return BadRequest("error while uploading file");
+                }
+            }
+
+            return Ok(offerImages);
+        }
+
+        [HttpPost("image/{offerSlug}"), Authorize(Roles = "User")]
```

Comment:
Odwrócił bym to dla zachowania hierarchii. To w końcu oferta ma zdjęcia, a nie na owdrót, czyli moim zdaniem `/{offerSlug}/image`

Reply:
Poprawiłem